test/credentials.mjs

```diff
@@ -1 +1,318 @@
-describe.skip("credentials", function () {});
+import * as nodeFs from "node:fs";
+
+import * as awilix from "awilix";
+import { expect } from "chai";
+import path from "path";
+import sinon, { spy } from "sinon";
+
+import { run } from "../src/cli.mjs";
+import { setupTestContainer as setupContainer } from "../src/config/setup-test-container.mjs";
+import { makeAccountRequest as originalMakeAccountRequest } from "../src/lib/account.mjs";
+import { runQueryFromString as originalRunQueryFromString } from "../src/lib/fauna-client.mjs";
+import { f } from "./helpers.mjs";
+
+const defaultAccountKeysFile = {
+  default: {
+    accountKey: "some-account-key",
+    refreshToken: "some-refresh-token",
+  },
+};
+const defaultDatabaseKeysFile = {
+  "some-account-key": {
+    "us-std:admin": {
+      secret: "some-database-secret",
+    },
+  },
+};
+
+// Ensure the credentials middleware correctly sets and refreshes account and database keys
+describe("credentials", function () {
+  let container, stderr, fs, fetch, credsDir, makeAccountRequest;
+
+  // Instead of mocking `fs` to return certain values in each test, let the middleware
+  //   read files in the actual filesystem.
+  const setCredsFiles = (accessKeys, secretKeys) => {
+    fs.mkdirSync(credsDir, { recursive: true });
+    fs.writeFileSync(
+      `${credsDir}/access_keys`,
+      JSON.stringify(accessKeys, null, 2),
+    );
+    fs.writeFileSync(
+      `${credsDir}/secret_keys`,
+      JSON.stringify(secretKeys, null, 2),
+    );
+  };
```

Comment:
:(

how much effort would it be to mock out these calls and have this function change the mock behaviors?

Reply:
Could be a decent chunk. Idea here was "set and forget" and let the middleware `get` and `save` as often as it needs to. Then at the end of the test I could assert that a certain key was used, knowing things were persisted correctly along the way. 

I wanted to avoid a function here that worries about `fs.readFileSync.onCall(x) -> spit out the new key instead of old one` etc.

Might be overthinking it :P